src/app/core/config/config.service.ts

```diff
@@ -0,0 +1,126 @@
+import { ConfigSuccessResponse, ErrorResponse, RestResponse } from '../cache/response-cache.models';
+import { ConfigRequest, FindAllOptions, RestRequest } from '../data/request.models';
+import { ResponseCacheEntry } from '../cache/response-cache.reducer';
+import { hasValue, isNotEmpty } from '../../shared/empty.util';
+import { HALEndpointService } from '../shared/hal-endpoint.service';
+import { ConfigData } from './config-data';
+
+@Injectable()
+export abstract class ConfigService extends HALEndpointService {
+  protected request: ConfigRequest;
+  protected abstract responseCache: ResponseCacheService;
+  protected abstract requestService: RequestService;
+  protected abstract linkName: string;
+  protected abstract EnvConfig: GlobalConfig;
+  protected abstract browseEndpoint: string;
+
+  protected getConfig(request: RestRequest): Observable<ConfigData> {
+    const [successResponse, errorResponse] =  this.responseCache.get(request.href)
+      .map((entry: ResponseCacheEntry) => entry.response)
+      .partition((response: RestResponse) => response.isSuccessful);
+    return Observable.merge(
+      errorResponse.flatMap((response: ErrorResponse) =>
+        Observable.throw(new Error(`Couldn't retrieve the config`))),
+      successResponse
+      .filter((response: ConfigSuccessResponse) => isNotEmpty(response) && isNotEmpty(response.configDefinition))
+      .map((response: ConfigSuccessResponse) => new ConfigData(response.pageInfo, response.configDefinition))
+      .distinctUntilChanged());
+  }
+
+  protected getConfigByIDHref(endpoint, resourceID): string {
+    return `${endpoint}/${resourceID}`;
+  }
+
+  protected getConfigSearchHref(endpoint, options: FindAllOptions = {}): string {
+    let result;
+    const args = [];
+
+    if (hasValue(options.scopeID)) {
+      result = `${endpoint}/${this.browseEndpoint}`;
+      args.push(`uuid=${options.scopeID}`);
+    } else {
+      result = endpoint;
+    }
+
+    if (hasValue(options.currentPage) && typeof options.currentPage === 'number') {
+      /* TODO: this is a temporary fix for the pagination start index (0 or 1) discrepancy between the rest and the frontend respectively */
+      args.push(`page=${options.currentPage - 1}`);
+    }
+
+    if (hasValue(options.elementsPerPage)) {
+      args.push(`size=${options.elementsPerPage}`);
+    }
+
+    if (hasValue(options.sort)) {
+      let direction = 'asc';
+      if (options.sort.direction === 1) {
+        direction = 'desc';
+      }
+      args.push(`sort=${options.sort.field},${direction}`);
+    }
+
+    if (isNotEmpty(args)) {
+      result = `${result}?${args.join('&')}`;
+    }
+    return result;
+  }
+
+  public getConfigAll(): Observable<ConfigData> {
+    return this.getEndpoint()
+      .filter((href: string) => isNotEmpty(href))
+      .distinctUntilChanged()
+      .map((endpointURL: string) => new ConfigRequest(endpointURL))
+      .do((request: RestRequest) => {
+        setTimeout(() => {
```

Comment:
These setTimeouts around `requestservice.configure()` calls are no longer necessary. They were the reason for the performance issues in described in [this comment](https://github.com/DSpace/dspace-angular/pull/180#issuecomment-339632224).

Reply:
I will remove